core-library/src/main/java/org/silverpeas/core/admin/space/SpaceInst.java

```diff
@@ -267,11 +258,24 @@ public String getFirstPageExtraParam() {
   }
 
   public List<SpaceInst> getSubSpaces() {
-    return Collections.unmodifiableList(subSpaces);
+    return data.safeRead((p, s, c) -> List.copyOf(s));
+  }
+
+  public void setData(final List<SpaceProfileInst> profiles, final List<SpaceInst> subSpaces,
+      final List<ComponentInst> components) {
+    data.reset();
+    data.manualWrite((p, s, c) -> {
+      p.addAll(profiles);
+      s.addAll(subSpaces);
+      c.addAll(components);
+    });
```

Comment:
Here, accordingly to the implementation, the data aren't set but more added to the existing ones. So, Shouldn't the setData method be renamed to addData so to avoid misunderstanding?